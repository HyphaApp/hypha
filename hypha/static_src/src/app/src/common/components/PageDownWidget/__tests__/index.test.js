import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import PageDownWidget from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });


describe("Test pagedownwidget component", () => {
  const text = "test name";
  const name = "test name";
  const label = "test label";
  const required = false;
  const helperProps = {};
  const value = "1";
  const id = "1"
  const init = { "a" : 1}

  const subject = shallow(<PageDownWidget
    id = {id}
    init = {init}
    text={text}
    name={name}
    label={label}
    required={required}
    value={value}
    helperProps={helperProps}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes", () => {
    expect(subject.find('.preview').length).toBe(1);
    expect(subject.children().length).toEqual(2)
  });

  test('render a pagedownwidget component', () => {
    expect(subject).toMatchSnapshot();
  });

});
