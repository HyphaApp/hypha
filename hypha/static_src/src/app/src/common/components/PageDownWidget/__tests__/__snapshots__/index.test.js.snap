// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test pagedownwidget component render a pagedownwidget component 1`] = `
<div>
  <TinyMCE
    helperProps={Object {}}
    id="1"
    init={
      Object {
        "a": 1,
      }
    }
    label="test label"
    name="test name"
    required={false}
    value="1"
  />
  <div
    className="preview"
    dangerouslySetInnerHTML={
      Object {
        "__html": "1",
      }
    }
  />
</div>
`;
